### PR TITLE
[4.0][com_plugins] convert to prepared statement 

### PR DESCRIPTION
**Pull request by alikon**
### Summary of Changes
use prepared statement for SQL

### Testing Instructions

test com_plugins

### Expected result

should work as before

### Actual result
N/A

### CODE REVIEW

File: administrator/components/com_plugins/Model/PluginsModel.php [hunk 1]
```diff
@@ -251,13 +256,15 @@ protected function getListQuery()
...
+			$query->where($db->quoteName('a.folder') . ' = :folder')
+				->bind(':folder', $folder);
 		}
 
 		// Filter by element.
 		if ($element = $this->getState('filter.element'))
 		{
-			$query->where('a.element = ' . $db->quote($element));
+			$query->where($db->quoteName('a.element') . ' = :element')
+				->bind(':element',  $element);
```
Comment (Quy) [inline administrator/components/com_plugins/Model/PluginsModel.php]:
```suggestion
				->bind(':element', $element);
```

File: administrator/components/com_plugins/Model/PluginsModel.php [hunk 2]
```diff
@@ -230,19 +231,23 @@ protected function getListQuery()
...
 		{
-			$query->where('a.enabled = ' . (int) $published);
+			$published = (int) $published;
+			$query->where($db->quoteName('a.enabled') . ' = :published')
+				->bind(':published', $published, ParameterType::INTEGER);
 		}
 		elseif ($published === '')
 		{
-			$query->where('(a.enabled IN (0, 1))');
+			$query->whereIn($db->quoteName('a.enabled') , [0, 1]);
```
Comment (Quy) [inline administrator/components/com_plugins/Model/PluginsModel.php]:
```suggestion
			$query->whereIn($db->quoteName('a.enabled'), [0, 1]);
```

File: administrator/components/com_plugins/Field/PluginorderingField.php [hunk 3]
```diff
@@ -53,8 +53,10 @@ protected function getQuery()
 				)
 			)
 			->from($db->quoteName('#__extensions'))
-			->where('(type =' . $db->quote('plugin') . 'AND folder=' . $db->quote($folder) . ')')
-			->order('ordering');
+			->where($db->quoteName('type')  . ' = ' . $db->quote('plugin'))
+			->where($db->quoteName('folder') . ' = :folder')
+			->bind(':folder', $folder)
```
Comment (Quy) [inline administrator/components/com_plugins/Field/PluginorderingField.php]: Move below `order`.

Comment (alikon) [reply administrator/components/com_plugins/Field/PluginorderingField.php]: why?

Comment (Quy) [reply administrator/components/com_plugins/Field/PluginorderingField.php]: I recall from @HLeithner it has to be in the last position. Your other PR #25386 has it this way.

Comment (Quy) [reply administrator/components/com_plugins/Field/PluginorderingField.php]: Or maybe the `bind` has to be after the `where`?

Comment (HLeithner) [reply administrator/components/com_plugins/Field/PluginorderingField.php]: ->bind() returns only a PreparableInterface which (maybe) doesn't have a Order() function.

Comment (alikon) [reply administrator/components/com_plugins/Field/PluginorderingField.php]: anyway moved below ;)

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on c5f826085d452574ce1abcf02bd0f1d721221921<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25413">issues.joomla.org/tracker/joomla-cms/25413</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**